### PR TITLE
Refactor to deno

### PR DESCRIPTION
**Pull request by keithamus**
This reworks this repo to use Deno instead of Node/npm as the main environment to test/build in.

Deno comes with test, lint, and formatter utilities letting us standardise on a single tool for everything. `deno fmt` also formats markdown and json, and can be used with `--check` to fail in CI if things aren't formatted.

We can still publish to npm by running `deno bundle mod.ts > index.js` which produces a single js file for consumption that Node can use.

Deno also has a coverage tool which uses v8 tooling to generate coverage info. We can use this to get coverage reports for the code.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -0,0 +1 @@
+deno 1.15.2
```
Comment (koddsson) [inline .tool-versions:1]: What is this file?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+interface Result {
```
Comment (koddsson) [inline index.d.ts:1]: Does `deno bundle` not generate declaration files for us?

**Review by koddsson — APPROVED** [review]
I think this makes a ton of sense.

File: index.d.ts (hunk 2 above)
Comment (keithamus) [reply index.d.ts:1]: Not yet. See https://github.com/denoland/deno/issues/3385

File: .tool-versions (hunk 1 above)
Comment (keithamus) [reply .tool-versions:1]: It's [used for `asdf`](https://asdf-vm.com/manage/configuration.html#tool-versions) which is a pluggable version manager for all things. I can remove it if you feel it unnecessary.

Comment (koddsson) [reply .tool-versions:1]: Nah, it's all good 👍🏻

**Pull request merged**